lib/school_code_generator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class SchoolCodeGenerator
+  MAX_CODE = 1_000_000
+
+  def self.generate
+    number = Random.new.rand(MAX_CODE)
+    code = format('%06d', number)
+
+    code.match(/(\d\d)(\d\d)(\d\d)/) do |m|
+      [m[1], m[2], m[3]].join('-')
+    end
```

Comment:
Minor: Is this formatting definitely something we want to store in the database, i.e. is it not just a presentational concern?

Reply:
I wondered the same thing but settled on storing it in this format as this is what we send to the Profile API when creating the school.